docs/src/modules/utils/generateMarkdown.js

```diff
@@ -144,12 +150,22 @@ function generatePropDescription(description, type) {
     }
   }
 
-  return `${deprecated}${jsDocText}${signature}`;
+  let notes = '';
+  if (isElementTypeAcceptingRefProp(type)) {
+    notes +=
+      '<br><sup>*</sup>[Needs to be able to hold a ref](/guides/composition/#Caveat with refs).';
+  }
+
+  return `${deprecated}${jsDocText}${signature}${notes}`;
 }
 
 function generatePropType(type) {
   switch (type.name) {
     case 'custom': {
+      if (isElementTypeAcceptingRefProp(type)) {
+        return `element type<sup>*</sup>`;
+      }
+
```

Comment:
Can be removed.

Reply:
How so? I would rather dispaly `element type` with a foot note instead of the unwieldy `elementTypeAcceptingRef`

Reply:
We can lose the asterisk in any case as discussed previously. You said you would enumerate the supported types. Perhaps you meant in the description rather than here?